Remove references to IBM Q from the README.md (#8686)
Qiskit/qiskit-terra: README.md

**Qiskit** is an open-source framework for working with noisy quantum computers at the level of pulses, circuits, and algorithms.

> The core component of Qiskit is `qiskit-terra`.

Reviewer: Do we want to add a sentence saying that e.g. the applications and other packages are built on it, or just leave as is?

Author: I would prefer leaving it as it is for now.

> The core component of Qiskit is `qiskit-terra`.

Reviewer 2: If we're going to give the full package name in code format, it might not hurt to say that it supplies the `qiskit` namespace - I think this is a fairly common source of confusion for new users (and experienced ones, sometimes).  Alternatively, we could avoid `` `qiskit-terra` `` and just write "Terra", similar to how we had it before?

Author: good point.
```suggestion
The core component of Qiskit is **Terra**.
```

Now that Qiskit is installed, it's time to begin working with Terra.

> We are ready to try out a quantum circuit example, which is simulated locally using Qiskit's `BasicAer`. This is a simple example that makes an entangled state.

Reviewer 3: If we're going to mention basicaer here I would call out that it's very limited and different from qiskit-aer. If you want to do real simulation work you should use qiskit-aer as basicaer is just for testing.

Author: done in 7055ed04ad0746801d4236a03ac545d2b5ee3774

## Creating Your First Quantum Program in Qiskit Terra

Now that Qiskit is installed, it's time to begin working with Qiskit. To do this
we create a `QuantumCircuit` object to define a basic quantum program.

```python
from qiskit import QuantumCircuit
qc = QuantumCircuit(2, 2)
qc.h(0)
qc.cx(0, 1)
qc.measure([0,1], [0,1])
```

This simple example makes an entangled state, also called a [Bell state](https://qiskit.org/textbook/ch-gates/multiple-qubits-entangled-states.html#3.2-Entangled-States-).

Once you've made your first quantum circuit, you can then simulate it.
To do this, first we need to compile your circuit for the target backend we're going to run
on. In this case we are leveraging the built-in `BasicAer` simulator. However, this
simulator is primarily for testing and is quite basic and very limited (as the name
implies). You should consider more sophisticated simulators, such as [`qiskit-aer`](https://github.com/Qiskit/qiskit-aer/),

> for any real simulation work.

Reviewer 3: I think it's probably better to leave it as `BasicAer` just to have this example continue working after aer is removed from the metapackage. If we didn't we'd basically have to say do `pip install qiskit qiskit-aer` to run this example

```
from qiskit import transpile
from qiskit.providers.basicaer import QasmSimulatorPy
backend_sim = QasmSimulatorPy()
transpiled_qc = transpile(qc, backend_sim)
```

After compiling the circuit we can then run this on the ``backend`` object with:

```
result = backend_sim.run(transpiled_qc).result()
print(result.get_counts(qc))
```

The output from this execution will look similar to this:

```python
{'00': 513, '11': 511}
```

For further examples of using Qiskit you can look at the example scripts in **examples/python**. You can start with
[using_qiskit_terra_level_0.py](examples/python/using_qiskit_terra_level_0.py) and working up in the levels. Also
you can refer to the tutorials in the documentation here:

https://qiskit.org/documentation/tutorials.html

### Executing your code on a real quantum chip

You can also use Qiskit to execute your code on a **real quantum processor** and not just a simulator.

> To do this it provides an abstraction layer that lets hardware vendors build packages that provde an
> interface to their hardware from Qiskit. Using these ``providers`` you can run any Qiskit code against
> a real quantum computer. Some examples of these provider packages for running on real hardware are:

Reviewer 4: ```suggestion
Qiskit provides an abstraction layer that lets users run quantum circuits on hardware from any
vendor that provides an interface to their systems through Qiskit. Using these ``providers`` you can run any Qiskit code against
real quantum computers. Some examples of published provider packages for running on real hardware are:
```

{'00': 513, '11': 511}
```

> A script is available [here](examples/python/ibmq/hello_quantum.py), where we also show how to run the same program on a real quantum computer via IBM Quantum.

Reviewer 3: We're probably going to remove the ibmq examples directory so we can probably just delete this line

Author: done in 9296ac8610054488a2858d2bf508d51778a7e26a

### Executing your code on a real quantum chip

You can also use Qiskit to execute your code on a
**real quantum chip**.

> In order to do so, you need to configure Qiskit with credentials for using a quantum computer cloud provider. For example, to connect it to your IBM Quantum account:

Reviewer 3: For this section I'd actually make it more general and less IBM specific at the beginning and only mention IBM as an example. Something like:

```suggestion
Qiskit enables you to run code on real quantum computers. To to do this it provides an abstraction
layer that lets hardware vendors build packages that provde an interface to their hardware from Qiskit.
Using these ``providers`` you can run any Qiskit code against a real quantum computer.

For example, to use Qiskit with IBM Quantum's cloud quantum computing service:
```

{'00': 513, '11': 511}
```

A script is available [here](examples/python/ibmq/hello_quantum.py), where we also show how to run the same program on a real quantum computer via IBM Quantum.

### Executing your code on a real quantum chip

You can also use Qiskit to execute your code on a

> **real quantum chip**. 

Reviewer 2: ```suggestion
**real quantum chip**.
```

> In order to do so, you need to configure Qiskit with credentials for using a quantum computer cloud provider. For example, to connect with IBM Quantum account:

Reviewer 2: ```suggestion
In order to do so, you need to configure Qiskit with credentials for using a quantum computer cloud provider. For example, to connect it to your IBM Quantum account:
```

#### Configure your IBM Quantum credentials

1. Create an _[IBM Quantum](https://quantum-computing.ibm.com)_ account if you haven't already done so.

2. Get an API token from the IBM Quantum website under _[Account settings](https://quantum-computing.ibm.com/account) > API Token_.

> <img src="docs/images/ibm_account1.png" width="30%"/><img src="docs/images/ibm_account2.png" width="40%"/>

Reviewer 2: You probably want to put a bit of horizontal space between these two images as well.

If possible, it would be neater if we could rescreenshot the two to:
- avoid the white background encroaching in the account dropdown
- make both images the same natural height (i.e. the same number of pixels tall), even if their widths are a bit different.

They'll sit next to each other on the page a bit more neatly if so.  These images also won't show up on the PyPI page, and they might be a bit temperamental on GitHub too because of the relative address.  One trick I've used in the past is to upload the images into a random issue, then use the permalink to those images in the readme instead.  The images don't end up checked into the repo, but the only way anyone would really see them anyway is on some online rendered version.

As an example, that's how the images in [this repository's readme](https://github.com/openqasm/openqasm.vim) are hosted - they're in openqasm/openqasm#239 (it was before we split the organisation off, so everything was still in a monorepo then).

Author: updated in 5cb64e623

>>> IBMQ.load_account()
```

> Those who do not want to save their credentials locally should use instead:

Reviewer 2: I think the former is clearer in this situation; "locally" can also mean "in memory" and you'd have to get the intent from the rest of the context.

Reviewer 2: Oh, I just saw above that you and Julien were talking about this.  I still think it's worth using "disk" rather than locally here, because the next line is really trying to distinguish persistent and volatile storage.

Author: I agree with the emphasis on "persistency". So changed in 38690e56f7b19b07d245fd7ccc859fd6766ef156

>>> IBMQ.load_account()
```

Those who do not want to save their credentials in a persistent file should use instead:

```python

> >>> from qiskit import IBMQ

Reviewer 3: Should we really have this in the qiskit-terra readme? Especially with the state of flux around ibm provider packages maybe we should just link to the readme of the provider package du jour and as we shouldn't want to worry about how to setup authentication or credentials as part of qiskit terra.

Author: I agree. However, let's keep it for now until the provider README gets released and updated.
